holmes/core/supabase_dal.py

```
@@ -298,6 +301,21 @@ def __init__(self, cluster: str):
         self.skill_hierarchy_cache = TTLCache(maxsize=1, ttl=hierarchy_ttl)
         self.lock = threading.Lock()
 
+    def __connect(self, options: ClientOptions):
+        relay_key = fetch_supabase_api_key(self.account_id, self.cluster)
+        for key in filter(None, (KEY_CACHE.pop("key", None), relay_key)):
+            try:
+                self.__login(key, options)
+                KEY_CACHE["key"] = key
+                return
+            except Exception as e:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)